apps/src/applab/dontMarshalApi.js

```diff
@@ -180,6 +180,27 @@ export function appendItem(
   }
 }
 
+function interpreterGetValue(object, key) {
+  return getValue(object, key, true);
+}
+
+export function getValue(object, key, calledWithinInterpreter) {
+  const objectContents = calledWithinInterpreter ? object.properties : object;
+  const keyData = key.data;
+  return objectContents[keyData].data;
+}
+
+function interpreterAddPair(object, key, value) {
+  return addPair(object, key, value, true);
+}
+
+export function addPair(object, key, value, calledWithinInterpreter) {
+  const newKey = key.data;
+  const objectContents = calledWithinInterpreter ? object.properties : object;
```

Comment:
Why do you need this? Shouldn't this always be called from the interpreter?

Reply:
Great question - I was hoping you'd know 😄  

I'm following the example of `appendItem` and going off of the comment at the top of dontMarshalApi: 
```
// APIs designed specifically to run on interpreter data structures without marshalling
// (valuable for performance or to support in/out parameters)
//
// dropletConfig for each of these APIs should be marked with dontMarshal:true

/*
 * The interpreter functions are exported under a single 'default' object
 *
 * The same functions are exported individually with an extra final parameter
 * named calledWithinInterpreter. When called outside of the interpreter without
 * the final parameter, the functions will behave as expected in a normal JS
 * environment and will not use interpreter-specific data structures.
 *
 * This allows us to share a single implementation for both the normal versions
 * used by exported apps and for the interpreter optimized versions.
 ```
Now that we aren't exporting apps, this whole business of not marshaling might be moot, but I don't really know.

Reply:
Here's the context for the other examples I was following: https://github.com/code-dot-org/code-dot-org/pull/29786

Reply:
If `dontMarshalApi` really is only for exporting apps, and since we don't plan to export apps, sounds like there are maybe 2 things to think about here: 

1.) I should update this PR with a way to add these new blocks with the regular api.

2.) Is @code-dot-org/starlabs tracking work to remove the export code?

Reply:
I talked with @madelynkasula about this and it turns out the dontMarshalApi is needed for Export (which is different than Expo). So even though Expo is not going to happen anymore, we still need all of this functionality